tee-worker/litentry/pallets/identity-management/src/lib.rs

```diff
@@ -129,17 +131,23 @@ pub mod pallet {
 			origin: OriginFor<T>,
 			who: T::AccountId,
 			key: UserShieldingKeyType,
-			parent_ss58_prefix: u16,
 		) -> DispatchResult {
 			T::ManageOrigin::ensure_origin(origin)?;
-			// we don't care about the current key
-			UserShieldingKeys::<T>::insert(&who, key);
 
-			let prime_id = Self::build_prime_identity(&who, parent_ss58_prefix)?;
+			let prime_id = Self::build_prime_identity(&who)?;
 			if IDGraphs::<T>::get(&who, &prime_id).is_none() {
-				let context = <IdentityContext<T>>::new(<frame_system::Pallet<T>>::block_number());
+				// TODO: shall we activate all available networks for the prime id?
```

Comment:
I would suggest not to do so. If we let the user to decide when to set shielding key, it's out of control. The user may not set any shielding key. 
What do you mean by **set "evm" networks for `Identity::Substrate`** ?

Reply:
I agree it needs to be discussed. To me it's more like a product/user experience issue, so I look for opinions of @kaylawangnow @jonalvarezz @mikuczi

One opposing view would be: when the user sets the shielding key, it's confusing that they need to select the identity type - at least less natural than linking an identity. If they want to change the network information of the prime account, they can use the `set_identity_web3network` at any time

Reply:
> I would suggest not to do so. If we let the user to decide when to set shielding key, it's out of control. The user may not set any shielding key.

@BillyWooo What I mean is that the setting user shielding key use cause should invole picking networks for prime identity by user explicitly.

> What do you mean by set "evm" networks for Identity::Substrate ?

@BillyWooo This code will result in substrate identity being linked to user's IdGraph with all web3 networks activated, even Ethereum, Polygon and BSC. Will it be fine?

```
let web3networks = get_all_web3networks()
    .try_into()
    .map_err(|_| Error::<T>::Web3NetworkLenLimitReached)?;
let context = <IdentityContext<T>>::new(
    <frame_system::Pallet<T>>::block_number(),
    web3networks,
);
Self::insert_identity_with_limit(&who, &prime_id, context)?;
```

Reply:
Currently people link identities via IDHub, where it comes down to two distinct categories: substrate or evm. So as long as F/E send the right types it's all good.

But yes from code logic we should have a sanity check here. A follow-up question is how to handle it? Do you simply throw out an error, or silently leave out the unmatched network types? It may be another user experience problem @kaylawangnow @jonalvarezz

Reply:
I currently throw out an error if the types don't match, because 99% of users (exception: IDHub team) will link it using IDHub so I don't expect it to go wrong. If it however does, it must be a bug

Reply:
> The more I think about it the more I'm convinced that we should let user decide when setting shielding key

This is a very fair point. My reasoning from a user perspective is that since:
1. The user can update their linked networks at anytime, including disabling previously activated networks, and
2. This happens only for the main (prime) account

It is alright. 

I'd suggest to however leave this to clients, i.e., don't do any automatic network activation here but accept a `web3network` vec with the list of networks to activate as part of setting the shielding key, if any. 

This enable clients to do it under the hood or to display the option to the user. IDHub, as requested by Product, will do it automatically at the beginning but it leaves the door open to change it at anytime.

Additionally, another reason to not perform any automatic activation here is that we are using setUserShieldingKey as a recovery method, e.g., you forgot your password. It wouldn't be nice to get networks activating as a side effect.

Kasper: If the IDHub goes with this automatic network linking when setting shielding key, we will make sure to let the user know what will happen so they know it upfront.

Reply:
> But yes from code logic we should have a sanity check here. A follow-up question is how to handle it? Do you simply throw out an error, or silently leave out the unmatched network types? 

Throwing an error if any of networks doesn't match sounds totally sane. A partial success would be more complex to properly handle. Make it fail out loud 

> I currently throw out an error if the types don't match, because 99% of users (exception: IDHub team) will link it using IDHub so I don't expect it to go wrong. If it however does, it must be a bug

Do we have the data to deny or confirm that IDHub is the only client out there using our Parachain? It is public, anyone could be using it and toying around with it. as far as I know, we are not monitoring it extensively 😄

Futhermore, I don't think it is healthy for us, as a public parachain, to think only in terms of IDHub. We should always use the word "clients" and tackle issues holistically as service provider. This is, to not minimize the impact of issues because we can translate and control them within the IDHub.

Reply:
> Do we have the data to deny or confirm that IDHub is the only client out there using our Parachain? 

No - it's completely open.
But if not using IDHub, probably only tech guys can construct the parameters correctly by looking up the code, if they are that far, they should know the handling of mismatching networks in our code.